python/tk_multi_workfiles/actions/context_change_action.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) 2020 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Action to create a change context.
+"""
+
+from sgtk import TankError
+from sgtk.platform.qt import QtGui
+
+from .file_action import FileAction
+from .action import Action
+
+
+class ContextChangeAction(Action):
+    """
+    This action changes the current engine context.
+    """
+
+    def __init__(self, environment):
+        """
+        Constructor.
+
+        :param environment: A WorkArea instance containing the context we will switch to.
+        """
+        Action.__init__(self, "Change Context")
+        self._environment = environment
+
+    def execute(self, parent_ui):
+        """
+        Perform a context change operation.
+
+        :param parent_ui: Parent dialog executing this action.
+        """
+
+        try:
+            # create folders and validate that we can save using the work template:
+            try:
+                # create folders if needed:
+                FileAction.create_folders(self._environment.context)
```

Comment:
Why the change from `create_folders_if_needed` to `create_folders`? I'm looking at the code and I understand the intent, but I don't understand it in this context (no pun intended 😜).

Reply:
If you don't actually have a template then the `create_folders_if_needed` method will error since it expects a template. With Mari you won't have a template, and if you are changing context even if you set it up in Maya, it doesn't necessarily make sense to expect a template either.

As far as I can gather from the `create_folders_if_needed` method, it is there to try and avoid folder creation if possible based on resolving templates. Since we can't rely on that, creating the folders without the check seems like the better option.

I had initially started to modify that method to check if there was a template, but I thought it probably just makes sense to call the `create_folders` instead.

Reply:
I guess going back to using `create_folders_if_needed` and then modifying it, would mean that if you say were using it in Maya where a template did exist, you could avoid the folder creation overhead, potentially. 
I'm happy to go with either approach.